modulus/datapipes/climate/era5_hdf5.py

```diff
@@ -157,7 +157,9 @@ def parse_dataset_files(self) -> None:
         self.logger.info(f"Getting file stats from {self.data_paths[0]}")
         with h5py.File(self.data_paths[0], "r") as f:
             # truncate the dataset to avoid out-of-range sampling
-            data_samples_per_year = f["fields"].shape[0] - self.num_steps * self.stride
+            data_samples_per_year = (
```

Comment:
So the idea here is that each rank should have the same number of samples? Maybe add a comment to make it more clear?

Reply:
That is correct! Good point about adding a comment. Let me do that